api/service.go

```diff
@@ -505,3 +509,25 @@ func contextsForServiceProvision(s *service.Service) []permTypes.PermissionConte
 		permission.Context(permTypes.CtxService, s.Name),
 	)
 }
+
+func filterServicesInstancesByTags(sInstances []service.ServiceInstance, tags []string) []service.ServiceInstance {
+	if len(tags) == 0 {
+		return sInstances
+	}
+
+	tagMap := make(map[string]struct{}, len(tags))
+	for _, tag := range tags {
+		tagMap[tag] = struct{}{}
+	}
+
+	filteredInstances := make([]service.ServiceInstance, 0, len(sInstances))
+	for _, service := range sInstances {
+		for _, sTag := range service.Tags {
+			if _, exists := tagMap[sTag]; exists {
+				filteredInstances = append(filteredInstances, service)
```

Comment:
nifty little hack to simplify the code:
```suggestion
				return append(filteredInstances, service)
```
you can then get rid of "break" and return nil (or an empty slice) at the end which makes it clearer 😄

Reply:
@arthurcgc nice point, but I think the desired behavior is to remove the break word, cause the filter need to check the full array